docs/modules/vector-search/pages/vector-collections.adoc

```diff
@@ -0,0 +1,498 @@
+
+[tabs]
+====
+Java::
++
+--
+[source,java]
+----
+VectorCollection<String, String> vectorCollection = VectorCollection.getCollection(hazelcastInstance, collectionName);
+CompletionStage<Void> resultDelete = vectorCollection.deleteAsync("1");
+CompletionStage<VectorDocument<String>> resultRemove = vectorCollection.removeAsync("2");
+----
+--
+====
+
+NOTE: These methods do not delete index nodes but mark them as deleted. This may impact search speed and memory usage. To permanently remove nodes from the index, you must run index optimization after deletion. See <<optimize-collection, optimize method>>
+
+== Similarity search
+
+Vector search returns objects with vectors that are most similar to that of the query, based on specified metrics.
+When submitting a search query, you have the option to include one vector values representing your query embeddings and specify a limit on the number of results to retrieve.
+
+=== Single-index vector collection search
+
+For similarity search, use the `searchAsync` method of the `VectorCollection`.
+[tabs]
+====
+Java::
++
+--
+[source,java]
+----
+CompletionStage<SearchResults<String, String>> results = vectorCollection.searchAsync(
+        VectorValues.of(new float[] {0, 0}),
+        SearchOptions.builder()
+            .limit(10)
+            .includeVectors()
+            .includeValue()
+            .build()
+);
+----
+--
+====
+
+=== Multi-index vector collection search
+
+For similarity search, use the `searchAsync` method of the `VectorCollection`, specifying the index name to search.
+[tabs]
+====
+Java::
++
+--
+[source,java]
+----
+CompletionStage<SearchResults<String, String>> results = vectorCollection.searchAsync(
+        VectorValues.of("index_name", new float[] {0, 0}),
+        SearchOptions.builder()
+            .limit(10)
+            .includeVectors()
+            .includeValue()
+            .build()
+);
+----
+--
+====
+
+=== Similarity search options
+Search parameters are passed as an argument `searchOptions` to the searchAsync method.
+.Search options
+[cols="1,2,3a",options="header"]
+|===
+|Option|Description|Example
+
+|limit
+|How many vectors should be returned in search result
+|
+
+[tabs]
+====
+Java::
++
+--
+[source,java]
+----
+var options = SearchOptions.builder().limit(10).build();
+----
+--
+====
+
+|includeValue
+|Include the user value in the search result.
+|
+
+[tabs]
+====
+Java::
++
+--
+[source,java]
+----
+var options = SearchOptions.builder().includeValue().build();
+----
+--
+====
+
+|includeVectors
+|Include the vectors values in the search result.
+|
+
+[tabs]
+====
+Java::
++
+--
+[source,java]
+----
+var options = SearchOptions.builder().includeVectors().build();
+----
+--
+====
+
+|hints
+|Extra hints for the search.
+|
+
+[tabs]
+====
+Java::
++
+--
+[source,java]
+----
+var options = SearchOptions.builder().hint(Hints.PARTITION_LIMIT, 1).build();
```

Comment:
hints are tricky and subject to change. We need to have a clear messaging here about them.

Reply:
What exactly details do you want to mention about the hints?

Reply:
add the following disclaimer:

```
WARNING: hints allow fine-tuning some aspects of search execution but are subject to change and may be removed in future versions.
```

Reply:
a1ea853e7d1f346965b2aea139b03c13410fee13